### ISSUE TITLE
Allow usage without npm in path.

### ISSUE THREAD

**Issue by Zoltu** [madskristensen/JSON-Intellisense#14]
I don't have NPM in my path and I believe this is the cause of the following error:

```
[8:31:58 PM] 'npm' is not recognized as an internal or external command,
[8:31:58 PM] operable program or batch file.
```

I would like a way to tell the tool where NPM is installed.  While I can't speak for everyone, my experience is that putting everything in the path is a unixy thing and while some Windows developers do the same, it is less common to throw everything in the path.  Most tooling allows the user to configure where binaries are stored if they can't be auto-located or found via registry keys.

**Comment by madskristensen** [comment]
By downloading NodeJS you automatically get npm on the PATH. This is a prerequisite of this extension

**Comment by Zoltu** [comment]
You don't get NodeJS in your path necessarily, depending on the context of the install and the context you are running Visual Studio in.  I *believe* that NodeJS installs NPM into the path for the context of the user installing it, which means if a different user runs Visual Studio NPM may not be in the path.  On top of that there is the question of running NodeJS installer as administrator and/or running Visual Studio as an administrator.

TL;DR: it is possible to have NodeJS installed without NPM (or even node) in your path.  I believe that tooling (especially on Windows) should be resilient to this and allow the user to provide a path for any dependencies, or the tool should provide the dependencies itself.

**Comment by madskristensen** [comment]
Then I have to ship node.exe and npm with the extension.

**Comment by Zoltu** [comment]
That would be one option, though it has its own Caveats (this is often the solution that Visual Studio takes, for example it ships with the TypeScript compiler and I believe the runtime to run it on).  However, I would be happy with just the ability to specify a path to node/npm, that way if I have it installed but not in the path I can still use the plugin.
